src/robusta/integrations/resource_analysis/memory_analyzer.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime, timedelta, tzinfo
+from typing import Optional
+
+from prometheus_api_client import PrometheusConnect
+
+from ..prometheus.utils import PrometheusDiscovery
+from ...core.model.env_vars import PROMETHEUS_REQUEST_TIMEOUT_SECONDS
+
+
+class MemoryAnalyzer:
+    def __init__(self, prometheus_url: str, prometheus_tzinfo: tzinfo):
+        if prometheus_url is None or prometheus_url == "":
+            prometheus_url = PrometheusDiscovery.find_prometheus_url()
+
+        self.prom = PrometheusConnect(url=prometheus_url, disable_ssl=True)
+        self.default_prometheus_params = {"timeout": PROMETHEUS_REQUEST_TIMEOUT_SECONDS}
+
+        self.prometheus_tzinfo = prometheus_tzinfo
+
+    def get_max_node_memory_usage_in_percentage(self, node_name: str, duration: timedelta) -> Optional[float]:
+        """
+        Gets the maximal memory usage (in percentage) for the node with the given node name, in the time range between now and now - duration.
+        :return: a float between 0 and 1, representing the maximal percentage of memory in use by the given node
+        """
+
+        # We estimate the node memory usage in percentage by the expression (1 - MemAvailable)/MemTotal.
+        # However, since the corresponding metrics contain only the name of the node exporter's instance, and not the node name,
+        # we perform a join of the instance name with the node_uname_info metric, in order to filter on the node name.
+
+        max_memory_usage_in_percentage = self._get_max_value_in_first_series_of_query(
```

Comment:
I'm not sure I fully understood the explanation here.
Regardless, in the built-in grafana dashboard the node cpu utilization query is:

instance:node_memory_utilisation:ratio{job="node-exporter", job="node-exporter", instance="$instance"}

Can this help here?

Reply:
I just thought that there's a calculated metric for node_memory_utilization, which is used in the grafana dashboard, which might be easier to use without calculations.
I'm not sure if both provide the exact same output.

Reply:
fixed